python/lsst/daf/butler/core/schema.py

```diff
@@ -21,351 +21,148 @@
-    Table('VisitRangeDatasetJoin', metadata,
-          Column('visit_begin', Integer, nullable=False),
-          Column('visit_end', Integer, nullable=False),
-          Column('camera_name', String, nullable=False),
-          Column('dataset_id', Integer, nullable=False, primary_key=True),
-          Column('registry_id', Integer, nullable=False, primary_key=True),
-          ForeignKeyConstraint(['visit_begin', 'visit_end', 'camera_name'], [
-              'VisitRange.visit_begin', 'VisitRange.visit_end', 'VisitRange.camera_name']),
-          ForeignKeyConstraint(['dataset_id', 'registry_id'], [
-              'Dataset.dataset_id', 'Dataset.registry_id'])
-          )
-
-AbstractFilterDatasetJoinTable = \
-    Table('AbstractFilterDatasetJoin', metadata,
-          Column('abstract_filter_name', String, nullable=False),
-          Column('dataset_id', Integer, nullable=False, primary_key=True),
-          Column('registry_id', Integer, nullable=False, primary_key=True),
-          ForeignKeyConstraint(['abstract_filter_name'], [
-              'AbstractFilter.abstract_filter_name']),
-          ForeignKeyConstraint(['dataset_id', 'registry_id'], [
-              'Dataset.dataset_id', 'Dataset.registry_id'])
-          )
-
-TractDatasetJoinTable = \
-    Table('TractDatasetJoin', metadata,
-          Column('tract_number', Integer, nullable=False),
-          Column('skymap_name', String, nullable=False),
-          Column('dataset_id', Integer, nullable=False, primary_key=True),
-          Column('registry_id', Integer, nullable=False, primary_key=True),
-          ForeignKeyConstraint(['tract_number', 'skymap_name'], [
-              'Tract.tract_number', 'Tract.skymap_name']),
-          ForeignKeyConstraint(['dataset_id', 'registry_id'], [
-              'Dataset.dataset_id', 'Dataset.registry_id'])
-          )
-
-PatchDatasetJoinTable = \
-    Table('PatchDatasetJoin', metadata,
-          Column('patch_index', Integer, nullable=False),
-          Column('tract_number', Integer, nullable=False),
-          Column('skymap_name', String, nullable=False),
-          Column('dataset_id', Integer, nullable=False, primary_key=True),
-          Column('registry_id', Integer, nullable=False, primary_key=True),
-          ForeignKeyConstraint(['patch_index', 'tract_number', 'skymap_name'], [
-              'Patch.patch_index', 'Patch.tract_number', 'Patch.skymap_name']),
-          ForeignKeyConstraint(['dataset_id', 'registry_id'], [
-              'Dataset.dataset_id', 'Dataset.registry_id'])
-          )
+metadata = None  # Needed to make disabled test_hsc not fail on import
+
+
+def iterable(a):
+    """Make input iterable.
+
+    There are three cases, when the input is:
+    - iterable, but not a string -> iterate over elements (e.g. [i for i in a])
+    - a string -> return single element iterable (e.g. [a])
+    - not iterable -> return single elment iterable (e.g. [a]).
+    """
+    if hasattr(a, 'encode'):
```

Comment:
I had a debate about this with @natelust. This should be faster.

Reply:
```
In [3]: x = "xxx"

In [4]: %timeit hasattr(x, "encode")
109 ns ± 0.692 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit isinstance(x, str)
102 ns ± 0.961 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
It doesn't look obviously different. I think we should go for the obvious implementation rather than looking for a side effect (and I can easily break yours by having a class with an encode method).